### PR TITLE
Opensuse openjdk package

### PR DESCRIPTION
**Pull request by atheiman**
I ran into this while working with jruby on the ruby_build cookbook. looks like the opensuse zypper packages available for openjdk dont match up with what this cookbook looks for.

```
dokken:/ # zypper install openjdk-8-jdk
Loading repository data...
Reading installed packages...
'openjdk-8-jdk' not found in package names. Trying capabilities.
No provider of 'openjdk-8-jdk' found.
Resolving package dependencies...

Nothing to do.
dokken:/ # zypper search openjdk
Loading repository data...
Reading installed packages...

S | Name                             | Summary                                                          | Type
--+----------------------------------+------------------------------------------------------------------+-----------
  | java-1_7_0-openjdk               | SUSE's implementation of the OpenJDK 7 runtime environment       | package
  | java-1_7_0-openjdk-accessibility | Accessibility connector for OpenJDK 7                            | package
  | java-1_7_0-openjdk-demo          | OpenJDK 7 Demos                                                  | package
  | java-1_7_0-openjdk-devel         | SUSE's implementation of the OpenJDK 7 Development Environment   | package
  | java-1_7_0-openjdk-headless      | OpenJDK 7 runtime environment without X, audio and video support | package
  | java-1_7_0-openjdk-javadoc       | OpenJDK 7 API Documentation                                      | package
  | java-1_7_0-openjdk-plugin        | Java Web Start and plugin implementation                         | package
  | java-1_7_0-openjdk-src           | OpenJDK 7 Source Bundle                                          | package
  | java-1_8_0-openjdk               | OpenJDK 8 Runtime Environment                                    | package
  | java-1_8_0-openjdk               | OpenJDK 8 Runtime Environment                                    | srcpackage
  | java-1_8_0-openjdk-accessibility | OpenJDK 8 accessibility connector                                | package
  | java-1_8_0-openjdk-demo          | OpenJDK 8 Demos                                                  | package
  | java-1_8_0-openjdk-devel         | OpenJDK 8 Development Environment                                | package
  | java-1_8_0-openjdk-headless      | OpenJDK 8 Runtime Environment                                    | package
  | java-1_8_0-openjdk-javadoc       | OpenJDK 8 API Documentation                                      | package
  | java-1_8_0-openjdk-plugin        | Java Web Start and plugin implementation                         | package
  | java-1_8_0-openjdk-src           | OpenJDK 8 Source Bundle                                          | package
```

### CODE REVIEW

**Comment by atheiman** [comment]
i had to make some gemfile updates because i was running into a berkshelf bug not getting the `apt` cookbook installed as a dependency, and I also had to make some foodcritic exceptions:

```
FC084: Deprecated Chef::REST class used: ./recipes/windows.rb:53
FC059: LWRP provider does not declare use_inline_resources: ./providers/alternatives.rb:1
FC059: LWRP provider does not declare use_inline_resources: ./providers/ark.rb:1
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/alternatives.rb:61
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/alternatives.rb:76
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/alternatives.rb:87
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/ark.rb:182
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/ark.rb:206
FC085: Resource using new_resource.updated_by_last_action to converge resource: ./providers/ark.rb:257
```

I dont know enough about this cookbook to correct those, but if you want me to make foodcritics suggested changes let me know.

**Comment by atheiman** [comment]
@erichelgeson @cattywampus @bai @shortdudey123 @JackBracken I saw you folks have some recent commits on this, mind taking a look at this?

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache-2.0'
 description       'Installs Java runtime.'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '1.50.0'
+version           '1.50.1'
```
Comment (majormoses) [inline metadata.rb]: In general the versioning is left to the maintainers post acceptance for several reasons:
- we may disagree on the impact of the change and therefore how its versioned, in this case I would say its a minor not a patch
- PRs are not a FIFO queue, as such versioning them prior to acceptance leads to needing to ask people to rebase more often. For example the current version is 2.2.1

**Comment by majormoses** [comment]
looks like you need a rebase to fix some merge conflicts.

**Comment by tas50** [comment]
I went to merge this in and it looks like the package names are actually present now. I'm going to close this out unless there's additional issues you run into.

**Pull request closed without merging**